substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ReflectionConfigurationParser.java

```diff
@@ -180,28 +173,17 @@ private void parseMethods(List<Object> methods, T clazz) {
     }
 
     private void parseMethod(Map<String, Object> data, T clazz) {
-        String methodName = null;
+        checkAttributes(data, "reflection method descriptor object", Collections.singleton("name"), Collections.singleton("parameterTypes"));
+        String methodName = asString(data.get("name"), "name");
         List<T> methodParameterTypes = null;
-        for (Map.Entry<String, Object> entry : data.entrySet()) {
-            String propertyName = entry.getKey();
-            if (propertyName.equals("name")) {
-                methodName = asString(entry.getValue(), "name");
-            } else if (propertyName.equals("parameterTypes")) {
-                methodParameterTypes = parseMethodParameters(clazz, methodName, asList(entry.getValue(),
-                                "Attribute 'parameterTypes' must be a list of type names"));
-                if (methodParameterTypes == null) {
-                    return;
-                }
-            } else {
-                throw new JSONParserException(
-                                "Unknown attribute '" + propertyName + "' (supported attributes: 'name', 'parameterTypes') in definition of method for class '" + delegate.getTypeName(clazz) + "'");
+        if (data.containsKey("parameterTypes")) {
+            methodParameterTypes = parseMethodParameters(clazz, methodName, asList(data.get("parameterTypes"),
```

Comment:
Pedantic: could avoid the extra `containsKey` by just doing `get` and checking the result `!= null`.

Reply:
Good point - I've removed the redundant map access